x/authz/authorizations.go

```diff
@@ -0,0 +1,38 @@
+package authz
+
+import (
+	"github.com/gogo/protobuf/proto"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+// Authorization represents the interface of various Authorization types implemented
+// by other modules.
+type Authorization interface {
+	proto.Message
+
+	// MsgTypeURL returns the fully-qualified Msg service method URL (as described in ADR 031),
+	// which will process and accept or reject a request.
+	MsgTypeURL() string
+
+	// Accept determines whether this grant permits the provided sdk.ServiceMsg to be performed, and if
+	// so provides an upgraded authorization instance.
+	Accept(ctx sdk.Context, msg sdk.ServiceMsg) (AcceptResponse, error)
```

Comment:
I really would prefer to merged https://github.com/cosmos/cosmos-sdk/pull/9139 first, is that okay? Because this needs to be updated to remove `sdk.ServiceMsg`

Reply:
OK, I started the review few days ago, but didn't finish.